src/basic/architecture.h

```diff
@@ -229,6 +230,9 @@ int uname_architecture(void);
 #    define native_architecture() ARCHITECTURE_ARC
 #    define LIB_ARCH_TUPLE "arc-linux"
 #  endif
+#elif defined(__loongarch64)
+#    define native_architecture() ARCHITECTURE_LOONGARCH64
+#    define LIB_ARCH_TUPLE "loongarch64-linux-gnu"
```

Comment:
This might need revision due to ongoing discussion at loongson/LoongArch-Documentation#25?

Reply:
Hmm, I don't see anything in that pull request diff that'd influence this triplet (and I can't read the discussion). What part do you have in mind?

Reply:
I mean the part touching the "multiarch tuple" section; in the latest revision of ABI spec and toolchain code, the ABI is to be explicitly marked in the tuples, like:

- `loongarch64-linux-gnuf64` for the LP64D ABI (basically means the same as RISC-V LP64D),
- `loongarch64-linux-gnuf32` for the LP64F ABI (ditto),
- `loongarch64-linux-gnusf` for the LP64S ABI (the soft-float ABI is marked with "S" in LoongArch).

Therefore, this tuple should be updated as well when the spec update gets merged; we might want to wait for gcc, for definitive values to use here, to reduce code churn.

Reply:
Yeah, I guess we should wait then.